src/ferc_xbrl_extractor/cli.py

```diff
@@ -76,139 +74,93 @@ def parse_main():
         "-a",
         "--archive-path",
         default=None,
-        type=str,
+        type=Path,
         help="Specify path to taxonomy entry point within a zipfile archive. This is a relative path within the taxonomy. If specified, `taxonomy` must be set to point to the zipfile location on the local file system.",
     )
     parser.add_argument(
         "-m",
         "--metadata-path",
         default=None,
-        type=str,
+        type=Path,
         help="Specify path to output metadata extracted taxonomy. Metadata will not be extracted if no path is specified.",
     )
     parser.add_argument(
         "--loglevel",
         help="Set log level (valid arguments include DEBUG, INFO, WARNING, ERROR, CRITICAL)",
         default="INFO",
     )
-    parser.add_argument("--logfile", help="Path to logfile", default=None)
+    parser.add_argument("--logfile", help="Path to logfile", type=Path, default=None)
 
     return parser.parse_args()
 
 
-def instances_from_zip(instance_path: Path) -> list[InstanceBuilder]:
-    """Get list of instances from specified path to zipfile.
-
-    Args:
-        instance_path: Path to zipfile containing XBRL filings.
-    """
-    allowable_suffixes = [".xbrl"]
-
-    archive = zipfile.ZipFile(instance_path)
-
-    # Read files into in memory buffers to parse
-    return [
-        InstanceBuilder(
-            io.BytesIO(archive.open(filename).read()), filename.split(".")[0]
-        )
-        for filename in archive.namelist()
-        if Path(filename).suffix in allowable_suffixes
-    ]
-
-
-def get_instances(instance_path: Path) -> list[InstanceBuilder]:
-    """Get list of instances from specified path.
-
-    Args:
-        instance_path: Path to one or more XBRL filings.
-    """
-    allowable_suffixes = [".xbrl"]
-
-    if not instance_path.exists():
-        raise ValueError(
-            "Must provide valid path to XBRL instance or directory" "of XBRL instances."
-        )
-
-    if instance_path.suffix == ".zip":
-        return instances_from_zip(instance_path)
-
-    # Single instance
-    if instance_path.is_file():
-        instances = [instance_path]
-    # Directory of instances
-    else:
-        # Must be either a directory or file
-        assert instance_path.is_dir()  # nosec: B101
-        instances = sorted(instance_path.iterdir())
-
-    return [
-        InstanceBuilder(str(instance), instance.name.rstrip(instance.suffix))
-        for instance in sorted(instances)
-        if instance.suffix in allowable_suffixes
-    ]
-
-
-def main():
-    """CLI for extracting data fro XBRL filings."""
-    args = parse_main()
-
+def run_main(
```

Comment:
In general I like having a function that handles all the actual post-parsing logic, so we can call this function from a non-CLI environment if we need to.